tmt/schemas/common.yaml

```diff
@@ -0,0 +1,137 @@
+---
+
+#
+# Common JSON Schema definitions
+#
+
+$id: /schemas/common
+$schema: https://json-schema.org/draft-07/schema
+
+definitions:
+  # https://tmt.readthedocs.io/en/stable/spec/tests.html#require
+  beakerlib_library:
+    allOf:
+      - $ref: "/schemas/common#/definitions/fmf_id_base"
+
+    additionalProperties: false
+
+    properties:
+      url: true
+      ref: true
+      path: true
+      name: true
+      nick:
+        type: string
+
+  # https://fmf.readthedocs.io/en/stable/concept.html#identifiers
+  fmf_id:
+    allOf:
+      - $ref: "/schemas/common#/definitions/fmf_id_base"
+
+    additionalProperties: false
+
+    properties:
+      url: true
+      ref: true
+      path: true
+      name: true
+
+  # helper used by beakerlib_library and fmf_id
+  fmf_id_base:
+    type: object
+    minProperties: 1
+    properties:
+      url:
+        type: string
+        # TODO: maybe also "format: uri"
+        # together with "pattern" to enforce http instead
+        # of all possible protocols?
+        pattern: "^https?://"
+
+      ref:
+        type: string
+
+      path:
+        type: string
+
+      name:
+        type: string
+        pattern: "^/"
+
+  # https://tmt.readthedocs.io/en/stable/spec/core.html#link
+  link:
+    anyOf:
+      - $ref: "common#/definitions/link_without_note"
```

Comment:
The `note` field is optional but the relation part is required. Can we express it here?

Reply:
I will try to come up with the solution